src/main/java/org/dice_research/opal/civet/metrics/DateFormatMetric.java

```diff
@@ -0,0 +1,123 @@
+ * 
+ * Finally, an average of the following 4 cases has to be computed for the final
+ * score: dataset: issued + modified, distributions: issued + modified
+ * 
+ * @author Aamir Mohammed
+ */
+public class DateFormatMetric implements Metric {
+
+	static List<String> correctFormat = new ArrayList<String>();
+	static {
+		correctFormat.add("YYYY");
+		correctFormat.add("YYYY-MM");
+		correctFormat.add("YYYY-MM-DD");
+		correctFormat.add("YYYY-MM-DDThh:mmTZD");
+		correctFormat.add("YYYY-MM-DDThh:mm:ssTZD");
+		correctFormat.add("YYYY-MM-DDThh:mm:ss.sTZD");
+	}
+
+	public static int checkDateFormat(String dct_issued) {
+		for (String string : correctFormat) {
+			SimpleDateFormat Format = new SimpleDateFormat(string);
+			try {
+				Format.parse(dct_issued.trim());
+				return 5;
+
+			} catch (ParseException pe) {
+				continue;
+			}
+		}
+		return 0;
+	}
+
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final String DESCRIPTIONS = "If the date format is in the correct format then give 5 stars else 0 star"
+			+ "Finally, an average of the following 4 cases has to be computed for the finalscore: dataset: issued + modified, distributions: issued + modified";
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+		int result = 0;
+		LOGGER.info("Processing dataset " + datasetUri);
+		Resource dataSet = model.createResource(datasetUri);
+		if (dataSet.hasProperty(DCTerms.issued)
+				&& !(dataSet.getProperty(DCTerms.issued).getObject().toString().isEmpty())) {
+			String dct_issued = dataSet.getProperty(DCTerms.issued).getObject().toString();
+			result += checkDateFormat(dct_issued);
+		} else {
+			result += 0;
+		}
+		if (dataSet.hasProperty(DCTerms.modified)
+				&& !(dataSet.getProperty(DCTerms.modified).getObject().toString().isEmpty())) {
+			String dateissued = dataSet.getProperty(DCTerms.modified).getObject().toString();
+			result += checkDateFormat(dateissued);
+		} else {
+			result += 0;
+		}
+		int countDistributions = 0;
+		StmtIterator distribution = model
+				.listStatements(new SimpleSelector(dataSet, DCAT.distribution, (RDFNode) null));
+		while (distribution.hasNext()) {
+			Statement distSetSentence = distribution.nextStatement();
+			Resource dist = distSetSentence.getObject().asResource();
+			if (dist.hasProperty(DCTerms.issued)
+					&& !(dist.getProperty(DCTerms.issued).getObject().toString().isEmpty())) {
+				String dateissued = dist.getProperty(DCTerms.issued).getObject().toString();
+				result += checkDateFormat(dateissued);
+			} else {
+				result += 0;
+			}
+			if (dist.hasProperty(DCTerms.modified)
+					&& !(dist.getProperty(DCTerms.modified).getObject().toString().isEmpty())) {
+				String dateissued = dist.getProperty(DCTerms.modified).getObject().toString();
+				result += checkDateFormat(dateissued);
+			} else {
+				result += 0;
+			}
+			countDistributions++;
+		}
+		return result / (2 * countDistributions + 2);
+	}
```

Comment:
I think instead of calculating the total number of distributions, we need to calculate the **total occurences of dct:modified and dct:issued**(both in dataset and distributions) because we are not evaluating a score for **Availability of date** but instead, we need to check whether all available dates have a correct format.

Reply:
it should be: result / (total occurences of dct:modified and dct:issued)